x/operator/keeper/abci.go

```diff
@@ -47,18 +54,18 @@ func (k *Keeper) UpdateVotingPower(ctx sdk.Context, avsAddr string) error {
 		}
 		// TODO: for now, we ignore the error when the price round is not found and set the price to 1 to avoid panic
 	}
+
 	// update the voting power of operators and AVS
+	isSnapshotChanged := false
+	votingPowerSet := make([]*operatortypes.OperatorVotingPower, 0)
 	avsVotingPower := sdkmath.LegacyNewDec(0)
-	// check if self USD value is more than the minimum self delegation.
-	minimumSelfDelegation, err := k.avsKeeper.GetAVSMinimumSelfDelegation(ctx, avsAddr)
-	if err != nil {
-		// this error is handled earlier when calling `GetAVSSupportedAssets`,
-		// so we don't set the voting power to zero here.
-		return err
-	}
-
+	hasOptedOperator := false
 	opFunc := func(operator string, optedUSDValues *operatortypes.OperatorOptedUSDValue) error {
+		if !hasOptedOperator {
+			hasOptedOperator = true
+		}
 		// clear the old voting power for the operator
+		lastOptedUSDValue := *optedUSDValues
 		*optedUSDValues = operatortypes.OperatorOptedUSDValue{
 			TotalUSDValue:  sdkmath.LegacyNewDec(0),
```

Comment:
Reduce unnecessary object creation: use sdkmath.**LegacyZeroDec**() instead of sdkmath.LegacyNewDec(0).

Reply:
Thanks, it will be fixed in the next commit.